Userland/Libraries/LibWeb/Loader/ResourceLoader.cpp

```diff
@@ -250,6 +255,61 @@ void ResourceLoader::load(LoadRequest& request, Function<void(ReadonlyBytes, Has
 
             auto const fd = file_or_error.value();
 
+            // When local file is a directory generate directory entries page
+            auto maybe_is_valid_directory = Core::Directory::is_valid_directory(fd);
+            if (!maybe_is_valid_directory.is_error() && maybe_is_valid_directory.value()) {
+                // Generate HTML contents entries list
+                auto lexical_path = LexicalPath(request.url().serialize_path());
+                Core::DirIterator dt(lexical_path.string(), Core::DirIterator::Flags::SkipParentAndBaseDir);
+                Vector<DeprecatedString> names;
+                while (dt.has_next())
+                    names.append(dt.next_path());
+                quick_sort(names);
+
+                StringBuilder contents;
+                contents.append("<table>"sv);
+                for (auto& name : names) {
+                    auto path = lexical_path.append(name);
+                    struct stat st;
+                    memset(&st, 0, sizeof(st));
+                    stat(path.string().characters(), &st);
```

Comment:
...Alternatively, `Core::System::stat()` provides a nicer API than calling the LibC function.

Reply:
You are right, this is a better API to use